generate-spec

```diff
@@ -249,6 +251,64 @@ if (file_exists($controllersDir)) {
 }
 
 $routes = [];
+foreach ($controllers as $controllerName => $stmts) {
+	$controllerClass = null;
+	/** @var Class_ $class */
+	foreach ($nodeFinder->findInstanceOf($stmts, Class_::class) as $class) {
+		if ($class->name->name === $controllerName . "Controller") {
+			$controllerClass = $class;
+			break;
+		}
+	}
+	if ($controllerClass === null) {
+		Logger::error($controllerName, "Controller '" . $controllerName . "' not found");
+		continue;
+	}
+
+	/** @var ClassMethod $classMethod */
+	foreach ($nodeFinder->findInstanceOf($controllerClass->stmts, ClassMethod::class) as $classMethod) {
+		$name = substr($class->name->name, 0, -10) . "#" . $classMethod->name->name;
+
+		/** @var AttributeGroup $attrGroup */
+		foreach ($classMethod->attrGroups as $attrGroup) {
+			foreach ($attrGroup->attrs as $attr) {
+				if ($attr->name->getLast() !== "Route") {
+					continue;
+				}
+
+				$key = null;
+				$args = [
+					"name" => $name,
+				];
+				for ($i = 0, $iMax = count($attr->args); $i < $iMax; $i++) {
+					$arg = $attr->args[$i];
+
+					if ($arg->name !== null) {
+						$argName = $arg->name->name;
+					} else {
+						$argName = ["type", "verb", "url", "requirements", "defaults", "root", "postfix"][$i];
+					}
```

Comment:
Here I implemented support for unnamed args we talked about in https://github.com/nextcloud/openapi-extractor/pull/24. I don't think it is a bad thing to support it, but I can also see that named args are much better for readability and maintainability. In the routes that I migrated on server I only used an unnamed arg for the type parameter because it is already clear from the const name.